src/formatDate.js

```diff
@@ -50,7 +50,38 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateInput = date;
+  const oldFormat = fromFormat;
+  const newFormat = toFormat;
+  const oldDelimiter = oldFormat.splice(3, 1);
+  const newDelimiter = newFormat.splice(3, 1);
+
+  const dateArray = dateInput.split(oldDelimiter);
+  const dateObj = {};
+  let resultDate = '';
+  const resultDateArray = [];
+
+  for (let i = 0; i <= oldFormat.length; i++) {
+    dateObj[oldFormat[i]] = dateArray[i];
+  }
+
+  if (newFormat.includes('YYYY') && (oldFormat.includes('YY'))) {
+    dateObj.YYYY = (dateObj.YY > 29 ? '19' : '20') + dateObj.YY;
+    delete dateObj.YY;
```

Comment:
Do you really need `dateObj`? Consider adding elements to the array straight away